test/fishjam/cluster/api_test.exs

```diff
@@ -0,0 +1,248 @@
+defmodule Fishjam.Cluster.ApiTest do
+  @moduledoc false
+
+  # These tests can only be run with `mix test.cluster.epmd` or `mix test.cluster.dns`.
+
+  use ExUnit.Case, async: false
+
+  @token Application.compile_env(:fishjam, :server_api_token)
+  @headers [Authorization: "Bearer #{@token}", Accept: "Application/json; Charset=utf-8"]
+  @post_headers @headers ++ [{"Content-Type", "application/json"}]
+  @nodes ["app1:4001", "app2:4002"]
+
+  @moduletag :cluster
+  @max_test_duration 400_000
+
+  setup do
+    if Mix.env() != :test_cluster do
+      raise "Load balancing tests can only be run with MIX_ENV=test_cluster"
+    end
+
+    # Delete all leftover rooms after each test
+    on_exit(fn ->
+      for node <- @nodes do
+        with {:ok, %HTTPoison.Response{status_code: 200, body: body}} <-
+               HTTPoison.get("http://#{node}/test/local/room", @headers) do
+          rooms = body |> Jason.decode!() |> Map.get("data")
+
+          for room <- rooms, do: HTTPoison.delete("http://#{node}/room/#{room["id"]}", @headers)
+        end
+      end
+    end)
+
+    {:ok, %{nodes: @nodes}}
+  end
+
+  @tag timeout: @max_test_duration
+  test "load-balancing and listing rooms in a cluster", %{nodes: [node1, node2]} do
```

Comment:
To be honest I feel like a lot is tested and happening here in that test.
What do you think maybe of dividing it into smaller tests some scenarios or alternatively comment before the next chunk of code and asserts whats gonna happen. 
Right now it's a huge bulk of asserts and you really need to read the whole block to see what's going on. 

Maybe some scenarios for separate tests
- One room added, 1 on one node 0 on the other one
- Two rooms added, they are split between nodes
- Two rooms, we delete one room 

So basically the same as you have here but split between tests so we have easier time reading it even if we add a few lines of code.

Reply:
I agree also we could think about more helper functions, which would wrap parts of logic and these new tests could be simply calling this one function.